runtime/moonbase/src/governance/origins.rs

```diff
@@ -39,47 +40,19 @@ pub mod custom_origins {
 		ReferendumKiller,
 	}
 
-	impl TryFrom<u16> for Origin {
+	impl TryFrom<String> for Origin {
 		type Error = ();
-		/// TrackId => Origin
-		fn try_from(value: u16) -> Result<Origin, ()> {
-			match value {
-				1 => Ok(Origin::WhitelistedCaller),
-				2 => Ok(Origin::GeneralAdmin),
-				3 => Ok(Origin::ReferendumCanceller),
-				4 => Ok(Origin::ReferendumKiller),
+		fn try_from(value: String) -> Result<Origin, ()> {
```

Comment:
I suggest to use `strum` derive macro `EnumString` to autogenerate this implementation:

https://docs.rs/strum/latest/strum/derive.EnumString.html